main.nf

```diff
@@ -1,234 +1,292 @@
+            --fastqscreen_databases '/path/to/databases' \\
+            --checkqc_config '/path/to/checkqc.config'
+
+    Mandatory parameters:
+        --run_folder                        The folder to quality check
+
+    Optional parameters:
+        --result_dir                        Path to write results (default: results)
+        --bcl2fastq_outdir                  Folder name to check for fastq.gz files and demultiplexing stats (default: Unaligned)
+        --checkqc_config                    Configuration file for CheckQC
+        --assets_dir                        Location of project assests (default: "\$baseDir/assets").
+        --config_dir                        Location of project configuration files (default: "\$baseDir/config").
+        --script_dir                        Location of project scripts (default: "\$baseDir/bin")
+
+        --help                              Print this help message.
+
+    Notes:
+        * Always quote paths that are parameters to nextflow e.g. '/path/to/file'
+
+    """
+}
+
+if (params.help || !params.run_folder){
+    helpMessage()
+    exit 0
+}
+
+workflow {
+
+    main:
+    get_run_folder(params.run_folder) | check_run_quality
+
+    publish:
+    check_run_quality.out.projectqc            to: "${params.result_dir}/multiqc_by_project"
+    check_run_quality.out.flowcellqc           to: "${params.result_dir}/multiqc_by_flowcell"
+
+}
+
+workflow.onComplete {
+	log.info ( workflow.success ? "\nDone! Open the reports in your browser.\n" : "Oops .. something went wrong." )
+}
+
+def get_run_folder(run_folder) {
+
+    Channel.value(file(run_folder))
+        .ifEmpty { "Error: No run folder (--run_folder) given."; exit 1 }
+
+}
+
+def get_project_and_reads(run_folder) {
+
+    Channel
+        .fromPath("${run_folder}/${params.bcl2fastq_outdir}/**.fastq.gz", maxDepth: 10 )
+        .filter( ~/.*_[^I]\d_001\.fastq\.gz$/ )
+        .ifEmpty { "Error: No fastq files found under ${run_folder}/ !\n"; exit 1 }
+        .map {
+            it.toString().indexOf('Undetermined') > 0 ?
+                ['NoProject', it] :
+                [(it.toString() =~ /^.*\/Unaligned\/([^\/]+)\/.*\.fastq\.gz$/)[0][1],it]
+        }
+
+}
+
+def combine_results_by_project (fastqc_results,fastq_screen_results) {
+
+    fastqc_results.mix(fastq_screen_results).groupTuple().map { it -> tuple(it[0],it[1][0].flatten(),it[1][1].flatten()) }
 
-params.additional_output_dir = ""
-if (params.additional_output_dir.length()) {
-    additional_output_dir = file(params.additional_output_dir)
 }
 
+workflow check_run_quality {
+
+    /* Workflow Graph
+
+    Runfolder -> InterOp                                -> MultiQCPerFlowcell
+        |     -> GetQCThresholds                        -> MultiQCPerFlowcell
+        |     -> GetMetaData                            -> MultiQCPerFlowcell + MultiQCPerProject
+        |
+        \ -> "Undeterminded/<Project>" -> FastQC        -> MultiQCPerFlowcell + MultiQCPerProject
```

Comment:
I first read "Undeterminded/<Project>" as a path and was confused, maybe another symbol can be used for "or"? Also: Undeterminded --> Undetermined

Reply:
I changed this to the command-line syntax style.